lib/ansible/modules/cloud/amazon/ecs_service.py

```diff
@@ -426,6 +435,33 @@ def main():
             module.fail_json(msg="Service still not deleted after "+str(repeat)+" tries of "+str(delay)+" seconds each.")
             return
 
+    elif module.params['state'] == 'deployed':
+        if not existing:
+            module.fail_json(msg="Service '"+module.params['name']+" not found.")
+        # it exists, so we should check if desiredCount and runningCount of the PRIMARY deployment matches
+        # in case of successful deploy, return info about the cluster deployed
+        # in case if deploy is failed, return the message from latest event
+        delay = module.params['delay']
+        repeat = module.params['repeat']
+        successful = False
+        if not module.check_mode:
+            for i in range(repeat):
+                existing = service_mgr.describe_service(module.params['cluster'], module.params['name'])
+                status = existing['status']
+                for deployment in existing['deployments']:
+                    if deployment['status'] == 'PRIMARY':
+                        desired_count = deployment['desiredCount']
+                        running_count = deployment['runningCount']
+                        break
+                if status == "ACTIVE" and desired_count == running_count:
+                    results['changed'] = True
```

Comment:
I think you are right. If you run `state=deployed` without actually changing the task def revision, it will still say `changed=True`.
I will confirm and revise this.